spyder/plugins/updatemanager/workers.py

```diff
@@ -193,25 +287,31 @@ class WorkerUpdate(BaseWorker):
     def __init__(self, stable_only):
         super().__init__()
         self.stable_only = stable_only
-        self.latest_release = None
-        self.update_available = False
+        self.asset_info = None
         self.error = None
         self.checkbox = False
         self.channel = None
 
-    def _check_update_available(
-        self,
-        releases: list[Version],
-        github: bool = True
-    ):
-        """Checks if there is an update available from releases."""
+    def _check_asset_available(self, release: None | Version):
+        """Checks if there is an asset available for release."""
+
+        # Get asset info from Github
+        releases = get_github_releases()
+
+        if release is not None:
+            # Only keep Github releases that are also available at channel url
```

Comment:
I don't understand this comment. It seems that you're only keeping releases that are lower than `release`. And that doesn't make sense to me because we should keep the ones higher than `release` to perform the update, shouldn't we?

Reply:
So this behavior is not changed from before this PR, so perhaps the comment is not very clear. The `release` here is only provided if Spyder is a standard conda install (not conda-based app); this is the latest release listed in the spyder conda channel (e.g. main or conda-forge).

We do not want to consider releases on Github as available updates if they are not also available on Spyder's install channel because we instruct the user to install it if the user does not choose to update to our conda-based installer, and we do not want to instruct the user to install a version from a channel that is not yet available.

If we can change this behavior if we like, but that is what we decided before.

Reply:
I've clarified this comment and expanded and clarified the docstring.